tests/components/octoprint/test_config_flow.py

```diff
@@ -0,0 +1,218 @@
+"""Test the OctoPrint config flow."""
+from unittest.mock import patch
+
+from homeassistant import config_entries, data_entry_flow, setup
+from homeassistant.components.octoprint.config_flow import CannotConnect
+from homeassistant.components.octoprint.const import DOMAIN
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry
+
+
+async def test_form(hass):
+    """Test we get the form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert not result["errors"]
+
+    with patch(
+        "homeassistant.components.octoprint.config_flow.validate_connection",
```

Comment:
Don't patch our code in the config flow. Patch the api client.

Reply:
Fixed